In [3]:
import pandas as pd
import numpy as np

df = pd.read_csv("DB.csv")

# Creating a Columns adding the Total Spent during the period of 5 Years.

In [4]:

df['Total 5 years period'] = df['Total Spending 2016'] + df['Total Spending 2017'] + df['Total Spending 2018'] + df['Total Spending 2019']+ df['Total Spending 2020']
cols = list(df.columns)
df = df[cols[0:3] + [cols[-1]]+cols[3:12]]
print(df)


                         Brand Name              Generic Name  \
0          1st Tier Unifine Pentips      Pen Needle, Diabetic   
1     1st Tier Unifine Pentips Plus      Pen Needle, Diabetic   
2                          Abacavir          Abacavir Sulfate   
3                          Abacavir          Abacavir Sulfate   
4                          Abacavir          Abacavir Sulfate   
...                             ...                       ...   
9989                         Zytiga       Abiraterone Acetate   
9990                          Zyvox                 Linezolid   
9991                          Zyvox  Linezolid In Dextrose 5%   
9992                          Zyvox  Linezolid In Dextrose 5%   
9993                          Zyvox  Linezolid In Dextrose 5%   

      Manufactor Name  Total 5 years period  Total Spending 2016  \
0     Owen Mumford Us          8.595306e+05         2.162739e+05   
1     Owen Mumford Us          1.702817e+06         3.780410e+05   
2     Major Pha

# What are the top 20 manufactors producing the most drugs? (top 20)

In [5]:
Number_drugs = df['Manufactor Name'].value_counts().sort_values(ascending=False)
Number_drugs.head(20)

Mylan              270
Teva USA           251
Sandoz             240
Amneal Pharmace    211
AHP                208
Aurobindo Pharm    206
Actavis/Teva       206
Major Pharmaceu    174
Mylan Instituti    161
Northstar Rx Ll    151
Lupin Pharmaceu    150
Zydus Pharmaceu    142
West-Ward/Hikma    135
Rising Pharm       135
Dr.Reddy'S Lab     131
Glenmark Pharma    119
Par Pharm.         116
Apotex Corp        114
Valeant/Bausch     107
Perrigo Co.        103
Name: Manufactor Name, dtype: int64

## Let's take a look at the Percentage

In [6]:

Number_drugs_Porc =(df['Manufactor Name'].value_counts(normalize = True) * 100)
Number_drugs_Porc.head(20)


Mylan              2.701621
Teva USA           2.511507
Sandoz             2.401441
Amneal Pharmace    2.111267
AHP                2.081249
Aurobindo Pharm    2.061237
Actavis/Teva       2.061237
Major Pharmaceu    1.741045
Mylan Instituti    1.610967
Northstar Rx Ll    1.510907
Lupin Pharmaceu    1.500901
Zydus Pharmaceu    1.420853
West-Ward/Hikma    1.350810
Rising Pharm       1.350810
Dr.Reddy'S Lab     1.310786
Glenmark Pharma    1.190714
Par Pharm.         1.160696
Apotex Corp        1.140684
Valeant/Bausch     1.070642
Perrigo Co.        1.030618
Name: Manufactor Name, dtype: float64

## Total of drug produced by the top 20:

In [7]:
Number_drugs_Porc.head(20).sum()

33.31999199519712

**It is easy to conclude that when the 33.32% of all the drug are produced by only 20 companies out of 900 manufactors, There is a unbalance that needs atencion in the industry.**

In [8]:
Manufactor_grp = df.groupby(['Manufactor Name'])


# What manufactors are collecting money the most? (top 20)

In [9]:
Spent_Manufactor= Manufactor_grp['Total 5 years period'].sum().sort_values(ascending=False)
Spent_Manufactor.head(20)

Manufactor Name
Novo Nordisk       3.831530e+10
Boehringer Ing.    3.217346e+10
Gilead Sciences    3.190075e+10
BMS Primarycare    3.147041e+10
Eli Lilly & Co.    3.081339e+10
Janssen Pharm.     2.965410e+10
Merck Sharp & D    2.820278e+10
Glaxosmithkline    2.786099e+10
Sanofi-Aventis     2.713355e+10
Pfizer US Pharm    2.667853e+10
Astrazeneca        2.651138e+10
Celgene/BMS        2.499334e+10
Abbvie US LLC      1.971530e+10
Allergan Inc.      1.870667e+10
Novartis           1.805327e+10
Amgen              1.547315e+10
Astellas Pharma    1.441836e+10
Mylan              9.818508e+09
Teva USA           9.775919e+09
Pharmacyclics,     9.616382e+09
Name: Total 5 years period, dtype: float64

In [10]:
Generic_grp = df.groupby(['Generic Name'])

In [11]:
#Generic_grp['Manufactor Name'].value_counts()

# Is there a monopoly when it comes to the pharmaceutical industry?

In [22]:
Final = pd.concat([Spent_Manufactor, Number_drugs, Number_drugs_Porc], axis='columns')
Final.rename(columns ={'Total 5 years period': 'Total Collected (5 years)', 'Manufactor Name': 'Different drugs produced'}, inplace=True)
Final.head(20)

Total Collected (5 years)  Different drugs produced  \
Novo Nordisk                  3.831530e+10                        36   
Boehringer Ing.               3.217346e+10                        31   
Gilead Sciences               3.190075e+10                        21   
BMS Primarycare               3.147041e+10                        20   
Eli Lilly & Co.               3.081339e+10                        51   
Janssen Pharm.                2.965410e+10                        26   
Merck Sharp & D               2.820278e+10                        66   
Glaxosmithkline               2.786099e+10                        50   
Sanofi-Aventis                2.713355e+10                        31   
Pfizer US Pharm               2.667853e+10                        91   
Astrazeneca                   2.651138e+10                        38   
Celgene/BMS                   2.499334e+10                        11   
Abbvie US LLC                 1.971530e+10                        35   
Allergan Inc.                 1.870667e+10                        53   
Novartis                      1.805327e+10                        60   
Amgen                         1.547315e+10                        28   
Astellas Pharma               1.441836e+10                         7   
Mylan                         9.818508e+09                       270   
Teva USA                      9.775919e+09                       251   
Pharmacyclics,                9.616382e+09                         1   

                 Different drugs produced  
Novo Nordisk                     0.360216  
Boehringer Ing.                  0.310186  
Gilead Sciences                  0.210126  
BMS Primarycare                  0.200120  
Eli Lilly & Co.                  0.510306  
Janssen Pharm.                   0.260156  
Merck Sharp & D                  0.660396  
Glaxosmithkline                  0.500300  
Sanofi-Aventis                   0.310186  
Pfizer US Pharm                  0.910546  
Astrazeneca                      0.380228  
Celgene/BMS                      0.110066  
Abbvie US LLC                    0.350210  
Allergan Inc.                    0.530318  
Novartis                         0.600360  
Amgen                            0.280168  
Astellas Pharma                  0.070042  
Mylan                            2.701621  
Teva USA                         2.511507  
Pharmacyclics,                   0.010006

**There is no way to afirm that there is actually a monopoly on the pharmaceutical industry, however it is clear that more competicion is needed, and even the intelectual propety is critial element for the invovation and the progress of humanity, when it comes to essencials some flexibility must be creates, Companies like "Astellas Pharma", "Celgene",  are able to profit by concentrairing in very expensive and exclusive medication.**